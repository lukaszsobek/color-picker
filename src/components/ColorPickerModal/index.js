import React from "react";

const ColorPickerModal = props => {
    const {
        isModalVisible,
        suggestedColors,
        getClickedColor,
        dataLoadedState
    } = props;

    if(!isModalVisible) {
        return null;
    }

    if(dataLoadedState === "loading") {
        return <div className="loading-message">Loading...</div>
    }

    if(dataLoadedState === "error") {
        return <div className="error-message">Error fetching suggestions.</div>
    }

    if(suggestedColors.length === 0) {
        return <div className="no-results">No matches.</div>
    }

    const SuggestedColorsList = suggestedColors.map((color, key) => {
        const { hex, name } = color;

        return (
            <div
                key={key}
                onClick={() => getClickedColor(name)}
                className="suggested-color"
            >
                <div
                className="color-preview"
                style={{ backgroundColor: "#" + hex }}
                />
                {name}
            </div>
        );
    });

    return (
        <div className="suggested-colors">
            { SuggestedColorsList }
        </div>
    );

}

export default ColorPickerModal;