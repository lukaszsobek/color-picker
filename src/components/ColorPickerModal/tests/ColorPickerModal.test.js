import ColorPickerModal from "../";

describe("ColorPickerModal", () => {

    it("returns null if !isModalVisible",() => {

        const props = {
            isModalVisible: false
        }
        expect(ColorPickerModal(props)).toBe(null);
    });

    it("displays 0 items with isModalVisible",() => {

        const props = {
            isModalVisible: true,
            suggestedColors: []
        }
        expect(ColorPickerModal(props)).toEqual([]);
    });

    it("displays 2 items with isModalVisible",() => {

        const props = {
            isModalVisible: true,
            suggestedColors: [1,2]
        }
        expect(ColorPickerModal(props)).toMatchSnapshot();
    });
});