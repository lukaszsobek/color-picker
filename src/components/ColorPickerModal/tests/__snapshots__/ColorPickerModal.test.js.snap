// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPickerModal displays \`loading...\` message if data not loaded 1`] = `
<div
  className="loading-message"
>
  Loading...
</div>
`;

exports[`ColorPickerModal displays \`no matches\` with 0 items 1`] = `
<div
  className="no-results"
>
  No matches.
</div>
`;

exports[`ColorPickerModal displays 2 items 1`] = `
<div
  className="suggested-colors"
>
  <div
    className="suggested-color"
    onClick={[Function]}
  >
    <div
      className="color-preview"
      style={
        Object {
          "backgroundColor": "#undefined",
        }
      }
    />
  </div>
  <div
    className="suggested-color"
    onClick={[Function]}
  >
    <div
      className="color-preview"
      style={
        Object {
          "backgroundColor": "#undefined",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorPickerModal returns null if !isModalVisible 1`] = `null`;
